packages/node/src/indexer/api.service.ts

```diff
@@ -90,77 +56,45 @@ export class ApiService implements OnApplicationShutdown {
     return this;
   }
 
-  getApi(): ApiPromise {
+  getApi(): Indexer {
     return this.api;
   }
-
-  async getPatchedApi(
-    block: SubstrateBlock,
-    runtimeVersion: RuntimeVersion,
-  ): Promise<ApiAt> {
-    this.currentBlockHash = block.block.hash.toString();
-    this.currentBlockNumber = block.block.header.number.toNumber();
-
-    const apiAt = (await this.api.at(
-      this.currentBlockHash,
-      runtimeVersion,
-    )) as ApiAt;
-    this.patchApiRpc(this.api, apiAt);
-    return apiAt;
+  getSafeApi(height: number): SafeAPIService {
+    return new SafeAPIService(this.api, height);
   }
+}
 
-  private redecorateRpcFunction<T extends 'promise' | 'rxjs'>(
-    original: RpcMethodResult<T, AnyFunction>,
-  ): RpcMethodResult<T, AnyFunction> {
-    const methodName = this.getRPCFunctionName(original);
-    if (original.meta.params) {
-      const hashIndex = original.meta.params.findIndex(
-        ({ isHistoric }) => isHistoric,
+export class SafeAPIService {
```

Comment:
The main goal of the safe api is to query contracts a the indexed block height. This doesn't currently seem to be possible.

Reply:
This type also needs to be set on the global in the types package